src/scripts/main.js

```diff
@@ -1,3 +1,101 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('#growth-table');
+  const appendRowButton = document.querySelector('.append-row');
+  const removeRowButton = document.querySelector('.remove-row');
+  const appendColumnButton = document.querySelector('.append-column');
+  const removeColumnButton = document.querySelector('.remove-column');
+
+  const MAX_ROWS = 10;
+  const MIN_ROWS = 2;
+  const MAX_COLUMNS = 10;
+  const MIN_COLUMNS = 2;
+
+  function updateButtonState() {
+    const rowCount = table.rows.length;
+    const columnCount = table.rows[0]?.cells.length || 0;
```

Comment:
Ensure that the table has at least one row and one cell initially to avoid potential errors when accessing `table.rows[0]`. Consider adding a check or initializing the table with a minimum structure.